misk-hibernate/src/test/kotlin/misk/hibernate/AggregationQueryTest.kt

```diff
@@ -0,0 +1,282 @@
+      |  sumI64() return element type must be out Number? for SUM aggregations, but was java.time.LocalDate""".trimMargin()
+    )
+  }
+
+  @Test fun `finds errors in projections with aggregated properties`() {
+    val ex = assertThrows<RuntimeException> {
+      primitiveTransacter.transaction { session ->
+        queryFactory.newQuery<WrongAggregatedProjectionQuery>()
+          .wrongAggregation(session)
+      }
+    }
+
+    assertThat(ex).hasMessage(
+      """
+      |java.lang.IllegalArgumentException: Query class misk.hibernate.WrongAggregatedProjectionQuery has problems:
+      |  misk.hibernate.WrongAggregatedProjection#average element type must be Double? for AVG aggregations, but was java.lang.Object
+      |  misk.hibernate.WrongAggregatedProjection#count element type must be Long? for COUNT aggregations, but was java.lang.Object
+      |  misk.hibernate.WrongAggregatedProjection#countDistinct element type must be Long? for COUNT_DISTINCT aggregations, but was java.lang.Object
+      |  misk.hibernate.WrongAggregatedProjection#max element type must be out Comparable? for MAX aggregations, but was java.lang.Object
+      |  misk.hibernate.WrongAggregatedProjection#min element type must be out Comparable? for MIN aggregations, but was java.lang.Object
+      |  misk.hibernate.WrongAggregatedProjection#sum element type must be out Number? for SUM aggregations, but was java.lang.Object""".trimMargin()
+    )
+  }
+
+  private fun seedData() {
+    movieTransacter.transaction { session ->
+      session.save(jurassicPark)
+      session.save(rocky)
+      session.save(starWars)
+      session.save(theColorPurple)
+      session.save(ferrari)
+      session.save(theBoysInTheBoat)
+    }
+    primitiveTransacter.transaction { session ->
+      session.save(DbPrimitiveTour(false, 9, 8, 7, 6, '5', 4.0f, 3.0))
+      session.save(DbPrimitiveTour(false, 10, 8, 7, 6, '5', 4.0f, 3.0))
+      session.save(DbPrimitiveTour(false, 11, 8, 7, 6, '5', 4.0f, 3.0))
+      session.save(DbPrimitiveTour(true, 0, 3, 4, 5, '6', 7.0f, 8.0))
+      session.save(DbPrimitiveTour(true, 1, 3, 4, 5, '6', 7.0f, 8.0))
+      session.save(DbPrimitiveTour(true, 2, 3, 4, 5, '6', 7.0f, 8.0))
+    }
+  }
+}
+
+class AggregationQueryTestModule : KAbstractModule() {
+  override fun configure() {
+    install(MiskTestingServiceModule())
+    install(DeploymentModule(TESTING))
+
+    val primitivesConfig = MiskConfig.load<RootConfig>("primitivecolumns", TESTING)
+    install(JdbcTestingModule(PrimitivesDb::class))
+    install(HibernateModule(PrimitivesDb::class, primitivesConfig.data_source))
+    install(object : HibernateEntityModule(PrimitivesDb::class) {
+      override fun configureHibernate() {
+        addEntities(DbPrimitiveTour::class)
+      }
+    })
+
+    val moviesConfig = MiskConfig.load<MoviesConfig>("moviestestmodule", TESTING)
+    val dataSourceConfig = moviesConfig.mysql_data_source
```

Comment:
Note that this test uses a MySQL data source instead of a Vitess data soource because there is (fixed) bug in some older versions of vitess which can't compare VARCHAR types in aggregations. Misk's VitessCluster pulls an image from 2020 (possibly v6?). This bug was fixed in 2021. We'll need to upgrade Misk's Vitess SHA in order to run this test under Vitess, but that would definitely be a separate change.

Reply:
The error that is thrown on the currently used image is `vtgate: types are not comparable: VARCHAR vs VARCHAR`